src/utils/BlockUtils.ts

```diff
@@ -101,7 +106,7 @@ export class BlockFinder {
    * @notice Gets the latest block whose timestamp is <= the provided timestamp.
    * @param {number} timestamp timestamp to search.
    */
-  public async getBlockForTimestamp(timestamp: number | string): Promise<Block> {
+  public async getBlockForTimestamp(timestamp: number | string, hints: BlockFinderHint): Promise<Block> {
```

Comment:
Do we want to make this optional as well? Right now we'd have to pass `getBlockForTimestamp(..., {})` to be type compliant. It may be convenient to provide a default.
```suggestion
  public async getBlockForTimestamp(timestamp: number | string, hints: BlockFinderHint = {}): Promise<Block> {
```